File: preprocess.js
const fs = require('fs');
let dataset = require('./dataset.json');

dataset = dataset.messages.map(message => `${message.author.name}: ${message.content}`);
// eslint-disable-next-line max-len, no-useless-escape
dataset = dataset.map(message => message.replace(/((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/g, '').replace(/[^\s\w!@()\-=_.?<>'";:|]/g, '')).filter(message => !(message === ''));

fs.writeFileSync('dataset.txt', dataset.join('\n'));

File: preprocess.py
import json
import re

with open(
	'data/dataset.json',
	encoding='ascii',
	errors='ignore'
) as f:
	dataset = json.load(f);

dataset = [f'{message["author"]["name"]}: {message["content"]}' for message in dataset['messages']]
dataset = [re.sub(r'((https?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&\'\(\)\*\+,;=.]+)', '', message) for message in dataset]
dataset = [re.sub(r'[^\w\s\-=_.?<>\'";:|]', '', message) for message in dataset]
dataset = list(filter(lambda message: message != '', dataset));

with open(
	'data/dataset.txt',
	'w',
	encoding='utf-8',
	newline='\n',
	errors='ignore'
) as f:
	f.write('\n'.join(dataset));
